### PR TITLE
Fix hooking parent and sub module with a single hook

### PR DESCRIPTION
**Pull request by pgayvallet**
Fix https://github.com/elastic/require-in-the-middle/issues/79

Fix the bug causing a hook used for both a module and some of its sub module to only hook into the parent module. 

```js
new Hook(['sub-module/foo', 'sub-module'], handler)
```

### CODE REVIEW

**Review by gsoldevila — APPROVED** [review]
LGTM!

File: index.js [hunk 1]
```diff
@@ -224,12 +224,21 @@ function Hook (modules, options, onrequire) {
...
-      if (hasWhitelist === true && modules.includes(moduleName) === false) {
-        if (modules.includes(fullModuleName) === false) return exports // abort if module name isn't on whitelist
+      let isWhitelistedSubmodule = false
+      if (hasWhitelist) {
+        // abort if module name isn't on whitelist
+        if (!modules.includes(moduleName) && !modules.includes(fullModuleName)) {
+          return exports
+        }
+
+        if (modules.includes(fullModuleName)) {
```
Comment (trentm) [inline index.js:234]:
```suggestion
        if (modules.includes(fullModuleName) && moduleName !== fullModuleName) {
```

This all looks good. I can't break it. :)

One thing that was niggling me a little bit is this one case. If `moduleName === fullModuleName` -- e.g. when doing `require('sub-module')` -- then this would result in (a) the `isWhitelistedSubmodule === true` being slightly misleading and (b) the `if (!isWhitelistedSubmodule) { ...` block below being skipped.  However, I can't come up with a case where skipping that block isn't just fine -- I expect `res === filename` every time that happens.

tl;dr: What you have here works, as best I can tell, but I think I'd prefer to have this small patch.

**Review by trentm — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (trentm) [reply index.js:234]: Let me know if you have thoughts on the suggestion. Then I can get a release out of this soon.
